### PR TITLE
waagent: T6475: Added waagent build instructions

### PR DESCRIPTION
**Pull request by zdc**
## Change Summary

Added waagent build instructions

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://vyos.dev/T6475

## Component(s) name

WALinuxAgent, waagent

## Proposed changes

We need a version newer than in the Debian repository.
This commit adds instructions to build a version from sid.

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-build/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: packages/waagent/Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,32 @@
+// Copyright (C) 2023 VyOS maintainers and contributors
```
Comment (sever-sever) [inline packages/waagent/Jenkinsfile]: Could you change the year to 2024?

Comment (zdc) [reply packages/waagent/Jenkinsfile]: Done.

**Review by sever-sever — APPROVED** [review]

File: packages/waagent/Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+// is not a typo! You need this underscore if the line immediately after the
+// @Library annotation is not an import statement!
+@Library('vyos-build@current')_
+
+def package_name = 'waagent'
+def pkgList = [
+    ['name': "${package_name}",
+     'scmCommit': 'debian/2.9.1.1-2',
+     'scmUrl': 'https://salsa.debian.org/cloud-team/waagent.git',
+     'buildCmd': 'sudo mk-build-deps --install --tool "apt-get --yes --no-install-recommends"; sudo apt-get install --yes --no-install-recommends dpkg-source-gitarchive; ../build.py'],
```
Comment (c-po) [inline packages/waagent/Jenkinsfile:28]: As there are no external patches to be applied, do we really need: `../build.py`?

Why not simply build the DEB using: `dpkg-buildpackage -uc -us -tc -b`?

**Review by c-po — CHANGES_REQUESTED** [review]

File: packages/waagent/Jenkinsfile (hunk 2 above)
Comment (zdc) [reply packages/waagent/Jenkinsfile:28]: It is just a common approach to build all new packages from me - it is easier to keep one standard script and logic for everything than sanitize and add features on demand. Consider this as a part of my build system. :)

But if you want to keep everything at a bare minimum, I have no objections to removing it.

**Review by c-po — APPROVED** [review]

File: packages/waagent/Jenkinsfile (hunk 2 above)
Comment (Apachez-) [reply packages/waagent/Jenkinsfile:28]: Who will make sure to remove this custom waagent once this issue have been resolved upstream (in future updates of this packages at Debian)?

Comment (c-po) [reply packages/waagent/Jenkinsfile:28]: Packe is removed by the maintainers from the VyOS repo if required

**Pull request merged**